lib/src/http/vm/request_mixin.dart

```diff
@@ -42,75 +42,76 @@ abstract class VMRequestMixin implements BaseRequest, CommonRequest {
   @override
   void cleanUp() {
     if (_isSingle) {
-      _client?.close();
+      _client.close();
     }
   }
 
   @override
   Future<Null> openRequest([Object? client]) async {
-    final HttpClient? httpClient = client as HttpClient?;
+    final httpClient = client as HttpClient?;
     if (httpClient != null) {
       _client = httpClient;
       _isSingle = false;
     } else {
       _client = HttpClient();
       _isSingle = true;
     }
-    _request = await _client!.openUrl(method!, uri);
+    _request = await _client.openUrl(method!, uri);
   }
 
   @override
   Future<BaseResponse> sendRequestAndFetchResponse(
       FinalizedRequest finalizedRequest,
       {bool streamResponse = false}) async {
-    finalizedRequest.headers.forEach(_request!.headers.set);
+    final request = _request!;
```

Comment:
does setting a variable pass by value and prevent us from aborting the call later?

Reply:
Since it's final, it shouldn't be a problem. It's just a place to store the same reference as _request so that we get the type promotion to non-nullable, but when abortRequest is called it will operate on the same request instance.